fe/src/main/java/org/apache/doris/mysql/privilege/PrivTable.java

```diff
@@ -52,28 +61,24 @@ public void addEntry(PrivEntry newEntry, boolean errOnExist, boolean errOnNonExi
             entries.add(newEntry);
             Collections.sort(entries);
             LOG.info("add priv entry: {}", newEntry);
+            return newEntry;
         } else {
             if (errOnExist) {
                 throw new DdlException("User already exist");
             } else {
-                if (!checkOperationAllowed(existingEntry, newEntry, "ADD ENTRY")) {
-                    return;
-                } else {
-                    if (existingEntry.isSetByDomainResolver() && newEntry.isSetByDomainResolver()) {
-                        existingEntry.setPrivSet(newEntry.getPrivSet());
-                        LOG.debug("reset priv entry: {}", existingEntry);
-                    } else if (existingEntry.isSetByDomainResolver() && !newEntry.isSetByDomainResolver()
-                            || !existingEntry.isSetByDomainResolver() && !newEntry.isSetByDomainResolver()) {
-                        mergePriv(existingEntry, newEntry);
-                        existingEntry.setSetByDomainResolver(false);
-                        LOG.info("merge priv entry: {}", existingEntry);
-                    }
-                    return;
+                checkOperationAllowed(existingEntry, newEntry, "ADD ENTRY");
+                if (existingEntry.isSetByDomainResolver() && newEntry.isSetByDomainResolver()) {
+                    existingEntry.setPrivSet(newEntry.getPrivSet());
+                    LOG.debug("reset priv entry: {}", existingEntry);
+                } else if (existingEntry.isSetByDomainResolver() && !newEntry.isSetByDomainResolver()
+                        || !existingEntry.isSetByDomainResolver() && !newEntry.isSetByDomainResolver()) {
```

Comment:
seems that it equal with `!newEntry.isSetByDomainResolver()`

Reply:
You are right...